internal/timeutil/util.go

```diff
@@ -20,11 +20,9 @@ import (
 )
 
 func GetTimeInMilliSeconds(duration time.Duration) int64 {
-	if duration == -1 {
```

Comment:
There are places that -1 is passed here. Removing this `if` does not seem correct